### ISSUE TITLE
Sync folder lost from settings

### ISSUE THREAD

**Issue by d7oc** [owncloud/client#11411]
Under certain currently unknown conditions the client loses the configuration for some sync connections.

No data is lost and the connection can manually be restored from a backup config:
- Quit the client
- create a backup of `%APPDATA%\ownCloud\owncloud.cfg` / `~/Library/Preferences/ownCloud/owncloud.cfg`
- open a backuo for example `%APPDATA%\ownCloud\owncloud.cfg.backup_20231129_142446_5.2.0.0`
- Locate the config for the sync that got lost
  - Here `1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\localPath=C:/Users/xxx/ownCloud - test@owncloud.server.com/` 
- Copy all related entries:
```
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\localPath=C:/Users/xxx/ownCloud - test@owncloud.server.com/
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\journalPath=.sync_journal.db
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\targetPath=/
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\davUrl=@Variant(\0\0\0\x11\0\0\0\x37https://owncloud.server.com/remote.php/dav/files/test/)
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\displayString=ownCloud
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\paused=false
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\ignoreHiddenFiles=true
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\deployed=false
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\priority=0
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\virtualFilesMode=wincfapi
1\Folders\b61f197e-333b-4ec4-b43e-3513ab03c5e3\version=13
```
- Insert them in  `%APPDATA%\ownCloud\owncloud.cfg` / `~/Library/Preferences/ownCloud/owncloud.cfg`
- Start the client

<details>
  <summary>Original report</summary>
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

The ownCloud Mac client started staying in paused mode since some time. I haven't found a solution yet. Logging out and in again didn't solve the issue.

### Expected behavior

Client should sync

### Steps to reproduce the issue

1. Start client against https://cloud.owncloud.com/
2. Wait for the client to sync

### Screenshots

<img width="280" alt="Screenshot 2023-11-30 at 15 14 46" src="https://github.com/owncloud/client/assets/102977884/45a3a7dc-65ca-425c-8509-b74393fb9b70">

### Logs

-> https://cloud.owncloud.com/f/6132439

### Client version number

ownCloud 5.1.2.12600 [9a0beb](https://github.com/owncloud/client/commit/9a0bebd8dc0bfd4099940e982eb6d84f2c08a904)
Libraries Qt 6.4.3, OpenSSL 3.1.2 1 Aug 2023
Using virtual files plugin: suffix
OS: macos-23.1.0
QPA: cocoa

</details>

**Comment by TheOneRing** [comment]
Please add a screenshot of your client, are there any warnings?

**Comment by TheOneRing** [comment]
But also no sync folder.

**Comment by d7oc** [comment]
The sync folder is there. There is also some sync activity shown in the in the `Server Activity` tab.

**Comment by hellerbarde** [comment]
I have observed this issue as well. Unfortunately I can't reliably recreate it (yet) but I'm noticing a pattern and I suspect it has to do with situations where my WiFi is flaky or not available during or right after bootup. 

I figured I should mention it just in case it's relevant. 

<details><summary>My Version Information</summary>
<pre>
ownCloud 5.2.0.12726 [f6cca1](https://github.com/owncloud/client/commit/f6cca1968d4fda84c01bd239b46371d7b0131278)
Libraries Qt 6.4.3, OpenSSL 3.1.2 1 Aug 2023
Using virtual files plugin: suffix
OS: pop-6.5.6-76060506-generic
QPA: xcb
</pre>
</details>

**Comment by d7oc** [comment]
Happened again here with

```
ownCloud 5.2.0.12726 f6cca1
Libraries Qt 6.4.3, OpenSSL 3.1.2 1 Aug 2023
Using virtual files plugin: suffix
OS: macos-23.2.0
QPA: cocoa
```

Logs are available internally: https://cloud.owncloud.com/f/6153507

**Comment by TheOneRing** [comment]
I'd recommend using a 5 daily build which includes https://github.com/owncloud/client/commit/2b3157658959a4db53886dc1dc19f726d118ad7b but this "fix" is just a random guess.

@d7oc this happened unrelated to a client update I assume?

**Comment by d7oc** [comment]
Yes it happened without a client update.

**Comment by codepoet80** [comment]
Closed my issue (11464) as a dupe, and was able to use the work-around, but just wanted to add here that this is happening on macOS as well. I can reliably reproduce, so let me know if I can help with logs or anything.

**Comment by michaelstingl** [comment]
@codepoet80 happens also with the 5.2.1?

**Comment by codepoet80** [comment]
Yes, as noted on my [Issue](https://github.com/owncloud/client/issues/11464), I'm on 5.2.1.13040 979bb9 Mac

**Comment by michaelstingl** [comment]
> however, it could have been _caused_ on an earlier version, and the update didn't magically fix it.

Yes, update doesn't fix it. If the sync connection disappeared pre-5.2.1, upgrade to 5.2.1 doesn't fix it.

> I haven't seen the issue pop up yet today -- although its usually a few reboots before it appears. I guess I'll update here if it re-occurs?

Yes, please keep an eye on it. Please make sure you have the debug log enabled in the background:
https://doc.owncloud.com/desktop/5.2/appendices/troubleshooting.html#logging-to-a-temporary-directory

**Comment by TheOneRing** [comment]
Fixed in 5.2.1
